### ISSUE TITLE
False positive with Dict passed to kwargs

### ISSUE THREAD

**Issue by mthuurne** [python/mypy#6092]
When I run mypy on the following code:
```py
from typing import Dict, Mapping

def f(*vargs: int, **kwargs: object) -> None:
	pass

d = {} # type: Dict[str, object]
f(**d)

m = {} # type: Mapping[str, object]
f(**m)
```
It reports:
```
7: error: Argument 1 to "f" has incompatible type "**Dict[str, object]"; expected "int"
```
Line 7 contains `f(**d)`.

I don't understand why mypy expects `int` as the type of an argument passed using `**`: as far as I know, there is no way for a value passed using `**` to be captured by an argument declared with `*`. Even if the passed map contains a key named `vargs`, its value would still end up in `kwargs`.

Note that arguments passed with `**` are not positional, so the error message stating "Argument 1" is either poorly worded or perhaps the result of the same problem that causes the false positive.

What is confusing to me is that this false positive only occurs if I pass a `Dict`, not a `Mapping`. However, Python will create a new dictionary instance for `kwargs` in either case, so only the key and value types of the expression after `**` should matter, I think, not the mapping type.

I'm using mypy 0.650 with Python 3.6.5.

**Comment by JukkaL** [comment]
This is clearly a false positive. Thanks for reporting this!

The bug has apparently been around for a long time.

**Comment by ilevkivskyi** [comment]
> The bug has apparently been around for a long time.

Yes, I think this is a duplicate of https://github.com/python/mypy/issues/1969
